lib/src/main/kotlin/com/swmansion/starknet/data/responses/TransactionReceipt.kt

```diff
@@ -6,22 +6,154 @@ import kotlinx.serialization.ExperimentalSerializationApi
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.json.JsonNames
 
+@Serializable
+sealed class CommonTransactionReceipt {
+    abstract val hash: Felt
+    abstract val actualFee: Felt
+}
+
+@Serializable
+sealed class TransactionReceipt : CommonTransactionReceipt() {
+    abstract override val hash: Felt
+    abstract override val actualFee: Felt
+    abstract val status: TransactionStatus
+    abstract val rejectionReason: String?
```

Comment:
In json rpc docs, it is called statusData, is there a chance it can be not null, when status is different than `rejected`?

Reply:
Honestly I'm not sure. I don't think it'd make sense to have extra status data if transaction just got accepted but it's not very explicit from the spec.